src/goto-symex/goto_symex_state.h

```diff
@@ -317,7 +315,6 @@ inline goto_statet::goto_statet(const class goto_symex_statet &s)
     source(s.source),
     propagation(s.propagation),
     atomic_section_id(s.atomic_section_id),
-    safe_pointers(s.safe_pointers),
```

Comment:
:pick: Ideally this removal would have gone in a different commit with an explanation of why not required.

Reply:
This PR is now composed of three commits:
1) Removing the `base_type_eq` comparison from `local_safe_pointerst`.
2) Small bits of cleanup in `local_safe_pointerst`.
3) Moving the storage from `goto_symex_statet` to `path_storaget`.